core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/regionserver/transactional/TrxTransactionState.java

```diff
@@ -126,15 +130,49 @@
 	    }
 	    catch (NoSuchMethodException exc_nsm2) {
 		sb_sqm_98_4 = false;
+                LOG.info("Got info of Class ScanQueryMatcher for HBase version" + VersionInfo.getVersion());
 	    }
 	}
-
 	if (sb_sqm_98_1) {
 	    LOG.info("Got info of Class ScanQueryMatcher for HBase 98.1");
 	}
 	if (sb_sqm_98_4) {
 	    LOG.info("Got info of Class ScanQueryMatcher for HBase 98.4");
 	}
+	
+	
+	Class scaninfoClazz = null;
```

Comment:
I believe scaninfoclazz should also be made static. Can you please check if the scanInfoConstructor becomes invalid if scanInfoclazz gets gc-ed.